content/blog/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages.md

```diff
@@ -0,0 +1,263 @@
+
+- builds my solution
+- creates a prerelease version of my package
+- creates a release version of my package
+- automatically publishes the prerelease package to a private Azure DevOps Artifacts feed
+- publishes the release version of the package to nuget.org if I give an approval
+
+In this post I'll go into how the pipeline is set up.
+
+First step is to enable the _Multi-stage pipelines_ preview feature:
+
+- Login to your Azure DevOps environment.
+- Click on your avatar in the top right corner.
+- Choose _Preview features_.
+- Enable the _Multi-stage pipelines_ option.  
+![Multi-stage pipelines preview feature](../../static/images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png)
+<!-- ![Multi-stage pipelines preview feature](../../../../../images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png) -->
+
+Now that the preview feature is enabled we can start. So, create a new pipeline and connect it to one of the sources that supports YAML, like Azure Repositories or GitHub. You can choose any pipeline template, because we're going to clear it and start from scratch. The [Build, test, and deploy .NET Core apps](https://docs.microsoft.com/azure/devops/pipelines/languages/dotnet-core) documentation is a good starting point if you haven't created a pipeline before.
+
+### Create NuGet packages
+
+Clear the YAML editor and add the following YAML. This will build any projects in your git repository using the .NET Core 2.2 SDK.
+
+```yaml
+trigger:
+- master
+
+stages:
+
+- stage: 'Build'
+  variables:
+    buildConfiguration: 'Release'
+
+  jobs:
+  - job:
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    workspace:
+      clean: all
+
+    steps:
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core sdk'
+      inputs:
+        packageType: sdk
+        version: 2.2.x
+        installationPath: $(Agent.ToolsDirectory)/dotnet
+    
+    - task: DotNetCoreCLI@2
+      displayName: "NuGet Restore"
+      inputs:
+        command: restore
+        projects: '**/*.csproj'
+    
+    - task: DotNetCoreCLI@2
+      displayName: "Build Solution"
+      inputs:
+        command: build
+        projects: '**/*.csproj'
+        arguments: '--configuration $(buildConfiguration)'
+```
+
+There are a few things to note. First the pipeline will trigger on a push to master. Next, we specify the `stages` keyword. Indicating that this is a multi-stage pipeline. In the first stage we'll build the solution and create the packages.
+
+My package targets _netstandard2.0_ so we're installing the .NET Core SDK as the first step in the stage. We're using the [UseDotNet](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops) task. It allows us to specify a wildcard for the version. Ensuring that we're always using the latest available version of the .NET Core SDK.
```

Comment:
De tweede zin is nogal kort. Ik zou ze cobineren. Since my pacakge targets _netstandard2.0_, we're installing the .NET Core SKD in the first step by using the [UseDotNet] task. This allows us to ....

Wat vind jij hiervan?